core/src/com/po/fuck/GeometryMisc.java

```diff
@@ -35,7 +36,21 @@ public static <A extends Entity, B extends Entity> float distance(A a, B b) {
     public static <A extends Entity, B extends Entity> Vector2 direction(A from, B to) {
         return to.position.cpy().sub(from.position);
     }
-
+    public static <A extends Entity> Entity findClosestOpponent(A entity) {
```

Comment:
I don't think that `GeometryMisc` should do queries to collections. You can do a function findClosest with `Entity` and a list of `Entities` to which you want to find it.